soco/core.py

```diff
@@ -1490,6 +1493,59 @@ def get_item_album_art_uri(self, item):
         else:
             return None
 
+    @only_on_master
+    def set_sleep_timer(self, sleep_time_seconds):
+        """Sets the sleep timer.
+
+        Args:
+            sleep_time_seconds (`int`): How long to wait before turning off
+                speaker in seconds. Maximum value of 86399
+
+        Returns:
+            bool: True if succesful, False otherwise
+
+        Raises SoCoException (or a subclass) upon errors, ValueError for
+            incorrect syntax.
+
+        """
+        if sleep_time_seconds and \
+           sleep_time_seconds != int(sleep_time_seconds):
+            raise ValueError('invalid sleep_time_seconds, must be integer \
+                value between 0 and 86399 inclusive')
+        if sleep_time_seconds:
+            # pylint: disable = redefined-variable-type
+            sleep_times = datetime.timedelta(seconds=int(sleep_time_seconds))
+            sleep_times = str(sleep_times).split(':')
```

Comment:
I'm not quite sure. Maybe this was what you referred to in your own comment. It looks a little redundant. You mentioned something about the ours being on digit or two. My point is, that if the device doesn't care then neither should we. If it does, there may be a cleverer way to do the formatting. I think on that a little.